app/code/Magento/Inventory/etc/di.xml

```diff
@@ -69,6 +69,28 @@
             </argument>
         </arguments>
     </type>
+
+    <virtualType name="Magento\Inventory\Model\Stock\SearchCriteria\StockCollectionProcessor" type="Magento\Framework\Api\SearchCriteria\CollectionProcessor">
+        <arguments>
+            <argument name="processors" xsi:type="array">
+                <item name="filters" xsi:type="object">Magento\Inventory\Model\Stock\SearchCriteria\CollectionProcessor\StockFilterProcessor</item>
+                <item name="sorting" xsi:type="object">Magento\Framework\Api\SearchCriteria\CollectionProcessor\SortingProcessor</item>
```

Comment:
Looks like we don't need to specify sorting and pagination (equals to default value)